src/drag-drop-context.tsx

```diff
@@ -1,30 +1,129 @@
-
-export const DragDropContext = (props) => {
-  props = mergeProps(
+import { createStore, Store } from "solid-js/store";
+
+type SensorActivator<K extends keyof HTMLElementEventMap> = (params: {
+  event: HTMLElementEventMap[K];
+  draggableId: string;
+}) => void;
+type Draggable = {
+  id: string;
+  node: HTMLElement;
+  layout: Layout;
+  data: Record<string, any>;
+  transform: Transform;
+};
+type Droppable = {
+  id: string;
+  node: HTMLElement;
+  layout: Layout;
+  data: Record<string, any>;
+  transform: Transform;
+};
+type DragEvent = {
+  draggable: Draggable;
+  droppable?: Droppable | null;
+};
+type RecomputeFilter = "all" | "draggable" | "droppable";
+
+interface Sensor {
+  id: string;
+  activators: { [K in keyof HTMLElementEventMap]?: SensorActivator<K> };
+}
+interface DragDropState {
+  draggables: Record<string, Draggable | null>;
+  droppables: Record<string, Droppable | null>;
+  sensors: Record<string, Sensor | null>;
+  active: {
+    draggable: string | null;
+    droppable: string | null;
+    sensor: string | null;
+  };
+  previous: { draggable: string | null; droppable: string | null };
+  usingDragOverlay: boolean;
+}
+interface DragDropActions {
+  setUsingDragOverlay(value?: boolean): void;
+  addDraggable(draggable: Omit<Draggable, "transform">): void;
+  addDroppable(droppable: Omit<Droppable, "transform">): void;
+  removeDraggable(id: string): void;
+  removeDroppable(id: string): void;
+  addSensor(sensor: Sensor): void;
+  removeSensor(id: string): void;
+  sensorStart(id: string): void;
+  sensorEnd(): void;
+  recomputeLayouts(filter?: RecomputeFilter): boolean;
+  detectCollisions(): void;
+  displace(
+    type: "draggables" | "droppables",
+    id: string,
+    transform: Transform
+  ): void;
+  activeDraggable(): Draggable | null;
+  activeDroppable(): Droppable | null;
+  activeSensor(): Sensor | null;
+  draggableActivators(draggableId: string, asHandlers?: boolean): Listeners;
+  anyDraggableActive(): boolean;
+  anyDroppableActive(): boolean;
+  anySensorActive(): boolean;
+  dragStart(draggableId: string): void;
+  dragMove(transform: Transform): void;
+  dragEnd(): void;
+  onDragStart(handler: DragEventHandler): void;
+  onDragMove(handler: DragEventHandler): void;
+  onDragOver(handler: DragEventHandler): void;
+  onDragEnd(handler: DragEventHandler): void;
+}
+interface DragDropContextProps {
+  onDragStart?: DragEventHandler;
+  onDragMove?: DragEventHandler;
+  onDragOver?: DragEventHandler;
+  onDragEnd?: DragEventHandler;
+  collisionDetectionAlgorithm?(
+    layout: Layout,
+    layouts: Layout[]
+  ): Layout | null;
+}
+
+type DragDropContext = [Store<DragDropState>, DragDropActions];
+type Listeners = Record<
+  string,
+  (event: HTMLElementEventMap[keyof HTMLElementEventMap]) => void
+>;
+type DragEventHandler = (event: DragEvent) => void;
+
+const Context = createContext<DragDropContext>();
+
+const DragDropProvider: Component<DragDropContextProps> = (passedProps) => {
```

Comment:
Overall, I'd say the naming should be using Provider as it's a practice seemingly recommended by Solid and more correctly describing what the component does (it provides the context to its children rather than being the context itself).
The alias can be a good move. With a bit more code it could also provide the older signature.

Reply:
Seeing as there will be other breaking changes, we can leave this as is and not worry about an alias I think (esp as pre-1.0).